### PR TITLE
[URP] Factorize template & SubTarget<UniversalTarget>

### PR DESCRIPTION
**Pull request by PaulDemeulenaere**
---
### Purpose of this PR
This PR follows up the initial refactor PR https://github.com/Unity-Technologies/Graphics/pull/3705, but adding for Sprite & Decal Targets:
- Use `UniversalSubTarget` instead of `SubTarget<UniversalTarget>`
- Use the über ShaderTemplate 

It will simplify the new ShaderGraph integration for VFX : https://github.com/Unity-Technologies/Graphics/pull/5448

---
### Testing status
Yamato ⏳ 

---
### Comments to reviewers
See also this [conversation](https://unity.slack.com/archives/GBJ7AU4V8/p1626854089328100)

### CODE REVIEW

File: com.unity.render-pipelines.universal/Editor/ShaderGraph/Targets/UniversalDecalSubTarget.cs [hunk 1]
```diff
@@ -74,8 +75,11 @@ public UniversalDecalSubTarget()
 
         public override bool IsActive() => true;
 
+        protected override ShaderUtils.ShaderID shaderID => ShaderUtils.ShaderID.SG_Decal;
+
         public override void Setup(ref TargetSetupContext context)
         {
+            base.Setup(ref context);
```
Comment (PaulDemeulenaere) [inline com.unity.render-pipelines.universal/Editor/ShaderGraph/Targets/UniversalDecalSubTarget.cs:82]: N.B.: Decal Sub Target still uses an independent template [DecalPass.template
](https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Editor/Decal/DecalPass.template) and it doesn't include the `kUberTemplatePath`.
It probably deserves to be factorized with the common template at some point.

**Comment by PaulDemeulenaere** [comment]
FYI: I moved the target from `master` to `HDRP/sg-vfx-integration-improved-to-urp` but I can move it back to `master` if there is any interest to include these change earlier in common master branch.

**Review by simon-engelbrecht-soerensen — APPROVED** [review]
Tested with Sprite & Decal shadergraphs, havent seen any regressions.

**Comment by PaulDemeulenaere** [comment]
I will do this later ⏳

**Review by pbbastian — APPROVED** [review]
LGTM

**Review by kennytann — APPROVED** [review]

**Pull request closed without merging**